### DISCUSSION TITLE
Saving additional online sources from MusicBrainz

### DISCUSSION THREAD

**Discussion by JOJ0** [beetbox/beets#4220] [Q&A] [answered]
And another question: I am wondering if it is possible to have both Musicbrainz IDs, but also Discogs IDs in the database.

As far as I understand now, by looking into the DB and using `beets info -l`, that if a track was tagged from Musicbrainz, it has all the information from MB obviously mb_albumid, mb_trackid, access to accousticbrainz things via the mb id's and so on.

If it is tagged from an release via the Discogs plugin, it seems to put the discogs_albumid, etc. fields but as well sets mb id fields to the same as the discogs id fields for some reason.

Anyway, my question is: Can I use both. I love musicbrainz/acousticbrainz because I get a lot of information about my music for DJ use, but on the other hand I am an oldschool music collector and have loads of vinyl as well, thus having direct access to Discogs release links for most of my files would be awesome.

musicbrainz does have direct links to discogs releases very often, I am aware of that, but is there a way to directly save it in the beets db alongside musicbrainz. E.g. thinking of things like displaying weblinks with a customized `beet ls` command or something.

Thanks in advance for your thoughts and tips!

**Comment by sampsyo** [comment]
We don't currently have a way to do save the Discogs ID from MusicBrainz. I imagine it wouldn't be the hardest thing to add, though, provided it comes along with the MB API response for releases by default.

**Comment by JOJ0** [comment]
Hmmm, I recall that I used musicbrainz discogs urls to match with my discogs collection over at my DiscoDOS project. It's somewhere here:
https://github.com/JOJ0/discodos/blob/master/discodos/model_brainz_match.py#L96

And yes it seems like it's included in the releases endpoint's result directly, no extra fetch, I seem to pull it out just from there:
https://github.com/JOJ0/discodos/blob/c4f114304b2e89ccbdc3dde41cbc6a5257422d4d/discodos/model_brainz.py#L108

Ah, but I see now that I actually include a bunch of extras in what the releases endpoint should give me:
https://github.com/JOJ0/discodos/blob/c4f114304b2e89ccbdc3dde41cbc6a5257422d4d/discodos/model_brainz.py#L76

(And btw don't blame me for that crappy hacky code, that could all be so much smarter)

**Comment by JOJ0** [comment]
I would love to look into it, maybe not this week but later on. Interested in how thinks work. Please point me to some starting points in the code and in the docs. Thanks so much in advance!
I am already reading this: https://beets.readthedocs.io/en/stable/dev/plugins.html

**Reply by sampsyo** [reply]
You'll want to start in the `beets.autotag.mb` module. For example, `track_info` is where we pull information out of an MB API response: https://github.com/beetbox/beets/blob/5d0442f0945fcd14d8783c76563e22cdbf735980/beets/autotag/mb.py#L196-L197

**Comment by arsaboo** [comment]
> If it is tagged from an release via the Discogs plugin, it seems to put the discogs_albumid, etc. fields but as well sets mb id fields to the same as the discogs id fields for some reason.

What happens when you tag it using MB? Is the `discogs_albumid` field still populated? The two ids look different, so maybe we can extend the Discogs plugin to check if it is a discogs id or MB id

**Reply by JOJ0** [reply]
As far as I rememeber (afk right now) discogs_albumid is not populated when the tagging source plugin is mb.

But as we discussed above, often mb saves a discogs link as well, grabbing it from the mb release response, theoretically just needs to be added to the mb tagger and both ids could be saved into beets! would be an addition i really would love and use heavily!!

**Comment by JOJ0** [comment, marked as answer]
PR #4708 which resolves this issue was just merged.
